### DISCUSSION TITLE
Flexible save image naming for professional workflows

### DISCUSSION THREAD

**Discussion by vicentecarro** [comfyanonymous/ComfyUI#2293] [Ideas]
Right now ComfyUI's save image node allows only for a prefix string that's prepended to the filename and then followed by a frame number. 
However now that AI Image Generation is becoming actually usable in production we need a more flexible naming in order to adapt to professional production workflows.
If you take for example an animation, VFX or similar workflow, you could expect to have naming that suits the fact that people work with scenes, shots, takes, sometimes versions of the takes (ugly), frame numbers, passes/layers/types/whatever( like normals, base, background, character, element, vfx, etx...), and sometimes some random 'ad hoc' stuff. 
So instead of the current comfyUI_08887.png naming pattern we would need something in the line of projectx_scene15_shot23_2_a_08887_vfx_smoke_base.png
As far as I know we don't have that functionality right now in ComfyUI, or in any other custom node that I know of. Could you include this much needed feature?
For starters I think ii would be enough to add a suffix field in the save image node, but later an specialized node could be added with independent fields for all elements, as that could help a lot to workflow designers/pipeline TDs/technical artists, etc.

**Comment by vicentecarro** [comment]
Turns out the node "Save Image Extended" can do something similar to this. That may be enough for a good share of use cases.

**Reply by NeedsMoar** [reply]
It has some pretty complex allowable syntax so it probably does...  you can use the detailed timestamping thing to output sequences to invidividual subfolders and such.   There's probably a way to rig one of the nodes that can switch between text items from a list to them if you're repeating words and I'm pretty sure it takes math expressions inline for the name fields for everything else, been a while since I looked though since it's overkill for my purposes
